File: core/src/nl/heretichammer/draculareignofterrorremake/component/PositionComponent.java
package nl.heretichammer.draculareignofterrorremake.component;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.math.Vector2;

public class PositionComponent extends Component {
	public final Vector2 position;
	
	public PositionComponent(Vector2 position) {
		this.position = position;
	}
	
	public PositionComponent(){
		this(new Vector2());
	}
}

File: src/components/PositionComponent.ts
import { Component, Types } from 'ecsy';
import { Vector2 } from '../math/Vector2';

export interface PositionComponentProps {
	position: Vector2;
}

export class PositionComponent extends Component<PositionComponentProps> {
	public position: Vector2;
}

PositionComponent.schema = {
	position: { type: Types.JSON },
};
